pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/FileDownloadService.java

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.dsm.service;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.google.auth.ServiceAccountSigner;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.storage.HttpMethod;
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.client.GoogleBucketClient;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.ddp.util.GoogleCredentialUtil;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+
+@Slf4j
+public class FileDownloadService {
+    private final ServiceAccountSigner signer;
+    private final GoogleBucketClient storageClient;
+    private final int maxSignedUrlMins;
+
+
+    public static FileDownloadService fromConfig(Config cfg) {
+        String signerJson = ConfigUtil.toJson(cfg.getConfig(ApplicationConfigConstants.FILE_DOWNLOAD_CREDENTIALS));
+        InputStream signerStream = new ByteArrayInputStream(signerJson.getBytes(StandardCharsets.UTF_8));
+        ServiceAccountCredentials signerCredentials;
+        try {
```

Comment:
`signerStream` is `InputStream` type and I believe we need to close it after its usage to avoid memory leakage, you could use `try-with-resource` to do close of streams automatically like: 
`try (signerStream) {
            signerCredentials = ServiceAccountCredentials.fromStream(signerStream);
        } catch (IOException e) {
            throw new DDPException("Could not get signer credentials", e);
        }`

Reply:
I believe `signerCredentials = ServiceAccountCredentials.fromStream(signerStream);` closes it automatically